### ISSUE TITLE
Citations don't use BetterBibTex keys

### ISSUE THREAD

**Issue by llrs** [rstudio/rstudio#10007]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2021.09.0 Build 351
    OS Version      : Ubuntu 20
    R Version       : 3.6.3 (and 4.1.0)

### Steps to reproduce the problem

I might miss something but I have a default Zotero 5.0.96.3 with BetterBibTex installed via the GUI. 
I installed the latest RStudio version. Under this setup go to a markdown edit it in visual mode (and after connecting to Zotero with BetterBibTex) add a reference to a document

### Describe the problem in detail

I  have BetterBibTex installed for Zotero but the configuration option to use BetterBibTex tag on the global options > R markdown > Citations does't appear. So the references I've been inserting don't use the same tag as I have on Zotero. 

I thought this was removed some time ago and used as default but I didn't find any mention of it being removed on the [RStudio release notes](https://www.rstudio.com/products/rstudio/release-notes/) and some documentation pages still show the option

![Screenshot from 2021-10-26 12-50-00](https://user-images.githubusercontent.com/6818218/138863448-f3fc8eb5-5608-4f55-afa3-bfffbd02076c.png)

### Describe the behavior you expected

I would expect to use the citation key of BetterBibTex. 

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by dragonstyle** [comment]
Hello - quick update for you. I've been able to reproduce this locally and am working on a fix that should get things back up and running. Thanks for reporting the issue and thanks for your patience!

**Comment by llrs** [comment]
Great to hear that! Thanks for fixing this. 
I'm not sure if the trouble to me to revert to use BBT keys as now there might be 150 citations without BBT keys...

**Comment by dragonstyle** [comment]
So sorry it took me so long to get to the bottom of it, but I appreciate you bringing it to our attention nonetheless. Hopefully you've saved some frustration for other users (or yourself on your next project). The issue should be fixed in an upcoming daily build, fixed as of this commit: 4892e3804acc9439f7037f166b911e780519e11f.

Thanks again.

**Comment by llrs** [comment]
No problem, I opened the issue for others too.  Should I close the issue or did you leave it open to remind to add tests?

**Comment by ronblum** [comment]
@llrs It's now in a `test` state to make sure that we verify the issue. However, are you seeing the correct behavior now?

**Comment by llrs** [comment]
Should install the daily version of rstudio to check it? Any way, I won't be able to test this until a couple of weeks.

**Comment by ronblum** [comment]
It would require installing one of the recent dailies. However, note that those are meant for testing purposes only, so it would be taking on a risk.

I'm marking this for internal verification, as well.

**Comment by ronblum** [comment]
Verified that the option is avaiable in 
- RStudio Desktop 2022.02.0+337

on
- Ubuntu 20.04
- MacOS 12.1
- Windows 10

**Comment by llrs** [comment]
I installed the rstudio-2022.02.0-daily-363 but it hang up when I clicked on the citation and tried to load a collection (and it didn't let me save for 10 minutes). I had to force closing RStudio and on reopening I had lost some changes I had done.  I had to revert back to rstudio-2021.09.1-372 so I couldn't really test it. 

Sorry but I won't risk redoing work on that project (too important) so I won't be able to investigate further the issue as I need to continue with my work.
